### PR TITLE
[SPARK-24151][SQL] Fix CURRENT_DATE, CURRENT_TIMESTAMP to be case insensitive

### PR DESCRIPTION
**Pull request by jamesthomp**
## What changes were proposed in this pull request?

This PR adjusts the behavior of CURRENT_DATE and CURRENT_TIMESTAMP to be case insensitive. This was previously the behavior prior to the merge of SPARK-22333.

## How was this patch tested?

Existing tests + added a new test to specifically catch this case.

### CODE REVIEW

**Comment by gatorsmile** [comment]
ok to test

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by gatorsmile** [comment]
@jamesthomp Could you document the behavior change in the migration guide? https://github.com/apache/spark/blame/master/docs/sql-programming-guide.md#L1802

**Comment by jamesthomp** [comment]
@gatorsmile - I have added a note to the migration guide.

File: docs/sql-programming-guide.md
Comment (felixcheung) [reply docs/sql-programming-guide.md]: we should port this correctness issue to branch 2.2, 2.3

Comment (dongjoon-hyun) [reply docs/sql-programming-guide.md]: +1 for @felixcheung 's suggestion.

Comment (HyukjinKwon) [reply docs/sql-programming-guide.md]: +1

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
@jamesthomp, mind updating this please?

**Comment by jamesthomp** [comment]
@HyukjinKwon - I have resolved the conflict with the docs changes. Please let me know if any additional changes are required.

File: docs/sql-programming-guide.md
Comment (dongjoon-hyun) [inline docs/sql-programming-guide.md]: Until now, 2.2.2 and 2.3.1 are released, too. Also, 2.3.2 voting is already started. 
So, the range seems to be `2.2.1 ~ 2.3.2`.

**Comment by dongjoon-hyun** [comment]
Hi, @robert3005 
I know you deleted the branch, but could you try this once more?
At this time, I can review and merge this if this is ready. Also, can we have a test in `AnalysisSuite` instead of `SQLQuerySuite`?

cc @gatorsmile

**Comment by dongjoon-hyun** [comment]
Ping once more since this can be merged into Spark 3.0, @robert3005 .

**Comment by HyukjinKwon** [comment]
Can anyone take over this then?

cc @kiszk, @mgaido91 and @viirya as well FYI.

**Comment by viirya** [comment]
@HyukjinKwon thanks for pinging me. I'd wait for others to take over this first, if no one does, I can do it later.

**Comment by mgaido91** [comment]
Thanks for pinging me @HyukjinKwon . I can take it over too, let me know. Thanks.

**Comment by jamesthomp** [comment]
I would be glad for @viirya or @mgaido91 to pick this up from me. It sounds like the remaining work is to move the test from `SQLQuerySuite` into `AnalysisSuite`, but I'm not certain how to easily do that as the form of the tests in `AnalysisSuite` is a bit different.
Feel free to take my code and push it up onto another branch, thanks!

**Comment by viirya** [comment]
Thanks @jamesthomp for you work. When we pick this up, I think we can still give the credit of the work to you.

**Comment by mgaido91** [comment]
if it's ok for you @viirya , I am submitting a PR for this then. I'll specify in the description that the credit should be given to @jamesthomp but this can be done also by the committer when eventually merging it I think. Thanks.

**Comment by viirya** [comment]
@mgaido91 No problem. Please submit the PR.

**Pull request closed without merging**
